search/binary_search1.cpp

```diff
@@ -0,0 +1,239 @@
+/**
+ * @file
+ * @brief [Binary search
+ * algorithm](https://en.wikipedia.org/wiki/Binary_search_algorithm)
```

Comment:
Missing a detailed description. You can take *some* of the detailed description from Wikipedia and adjust it to match the Doxygen documentation, otherwise, it would be plagiarism. 🙂